src/ray/protobuf/autoscaler_state_service.proto

```diff
@@ -0,0 +1,123 @@
+syntax = "proto3";
+option cc_enable_arenas = true;
+
+package ray.rpc;
+
+import "src/ray/protobuf/intance_manager.proto";
+// ============= Cluster Resources ====================
+//
+// Following fields represents the Cluster Resources autoscaler interested
+// in.
+
+// Represents an anti-affinity constraint. A bundle with this constraint
+// can't be allocated to a node that has a label with the same name and
+// value.
+message AntiAffinityConstraint {
+  string label_name = 1;
+  string label_value = 2;
+}
+
+message PlacementConstraint {
+  oneof constraint {
+    AntiAffinityConstraint anti_affinity = 1;
+  }
+}
+
+message ResourceBundle {
+  map<string, double> resources = 1;
+  PlacementConstraint placement_constraint = 2;
+  // If the resource bundle is allocated, the node will be labeled
+  // with the following labels.
+  // map<string, string> new_labels = 3;
+}
+
+// All bundles in the same resource request require gang
+// allocation semantics: they should be allocated together.
+message GangSchedulingRequest {
+  // a map from bundles to the number of bundles requested.
+  map<ResourceBundle, int32_t> bundles = 1;
+}
+
+// Cluster constraint represents minimial cluster size requirement,
+// issued through ray.autoscaler.sdk.request_resources.
+message OnDemandResourceRequest {
+  bytes request_id = 1;
+  GangSchedulingRequest resource_request = 2;
+}
+
+message NodeResourceState {
+  bytes node_id = 1;
+
+  // The address of the node.
+  string address = 2;
```

Comment:
Is this needed? Consider removing. Having access to address has only been a footgun in the past.